src/contrib/profiling-sssd.rst

```diff
@@ -0,0 +1,116 @@
+Profiling SSSD
+##############
+
+Getting Ready
+*************
+
+There are several tools allowing us to profile an application. We will focus on
+perf_ which seems to be the one that works best with SSSD.
+
+.. _perf: https://perf.wiki.kernel.org
+
+First you need to install the ``perf`` tool:
+
+.. code-tabs::
+
+    .. fedora-tab::
+
+        # dnf -y install perf js-d3-flame-graph
+
+    .. ubuntu-tab::
+
+        $ sudo apt install linux-tools-common linux-tools-generic
+
+
+Make sure the debug information is available for SSSD and all its dependencies
+either by using ``debuginfod``, or by manually installing the debug information
+packages:
+
+.. code-tabs::
+
+    .. fedora-tab::
+
+        # dnf -y --enable-repo=fedora-debuginfo --enable-repo=updates-debuginfo \
+        install sssd*debuginfo libsss_*-debuginfo samba-client-libs-debuginfo \
+        libldb-debuginfo libtevent-debuginfo libtalloc-debuginfo glibc-debuginfo
+
+    .. ubuntu-tab::
+
+        $ sudo apt install sssd-ad-common-dbgsym sssd-ad-dbgsym sssd-common-dbgsym \
+        sssd-dbus-dbgsym sssd-ipa-dbgsym sssd-kcm-dbgsym sssd-krb5-common-dbgsym \
+        sssd-krb5-dbgsym sssd-ldap-dbgsym sssd-proxy-dbgsym sssd-tools-dbgsym \
+        libsss-certmap0-dbgsym libsss-idmap0-dbgsym libsss-nss-idmap0-dbgsym \
+        libsss-simpleifp0-dbgsym libsss-sudo-dbgsym libldb2-dbgsym libtdb1-dbgsym \
+        libtalloc2-dbgsym libtevent0-dbgsym libc6-dbgsym
+
+.. seealso::
+
+    More information on `Debug Symbol Packages`_ for Ubuntu.
+
+.. _`Debug Symbol Packages`: https://documentation.ubuntu.com/server/reference/debugging/debug-symbol-packages/
+
+.. note::
+
+    If you are using SELinux, the shipped targeted policy may prevent ``perf``
+    from interacting with the SSSD process. You can put SELinux into permissive
+    mode to confirm or look at the audit logs and add the required rules. Rules
+    can be added using ``audit2allow``. 
```

Comment:
Technically, audit2allow generates a module that can be loaded. That detail is not important.

Reply:
I think we can trust that users adding SELinux rules know what the do. Non-expert users will probably set it to permissive mode while profiling.